samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/Application.java

```diff
@@ -0,0 +1,85 @@
+import com.microsoft.bot.integration.Configuration;
+import com.microsoft.bot.integration.spring.BotController;
+import com.microsoft.bot.integration.spring.BotDependencyConfiguration;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+
+/**
+ * This is the starting point of the Sprint Boot Bot application.
+ */
+@SpringBootApplication
+
+// Use the default BotController to receive incoming Channel messages. A custom
+// controller could be used by eliminating this import and creating a new
+// org.springframework.web.bind.annotation.RestController.
+// The default controller is created by the Spring Boot container using
+// dependency injection. The default route is /api/messages.
+@Import({BotController.class})
+
+/**
+ * This class extends the BotDependencyConfiguration which provides the default
+ * implementations for a Bot application.  The Application class should
+ * override methods in order to provide custom implementations.
+ */
+public class Application extends BotDependencyConfiguration {
+
+    /**
+     * The start method.
+     *
+     * @param args The args.
+     */
+    public static void main(String[] args) {
+        SpringApplication.run(Application.class, args);
+    }
+
+    /**
+     * Returns the Bot for this application.
+     *
+     * <p>
+     * The @Component annotation could be used on the Bot class instead of this method with the
+     * @Bean annotation.
+     * </p>
+     *
+     * @return The Bot implementation for this application.
+     */
+    @Bean
+    public Bot getBot(
+        Configuration configuration,
+        UserState userState,
+        ConversationState conversationState
+    ) {
+        TelemetryConfiguration telemetryConfiguration = new TelemetryConfiguration();
+        telemetryConfiguration.setInstrumentationKey(configuration.getProperty("InstrumentationKey"));
+        telemetryConfiguration.setChannel(configuration.getProperty("Channel"));
+        TelemetryClient telemetryClient = new TelemetryClient(telemetryConfiguration);
+        BotTelemetryClient botTelemetryClient = new ApplicationInsightsBotTelemetryClient(telemetryClient);
+
+        FlightBookingRecognizer recognizer =  new FlightBookingRecognizer(configuration, botTelemetryClient);
+        MainDialog dialog = new MainDialog(recognizer, new BookingDialog(), botTelemetryClient);
+        return new DialogAndWelcomeBot<>(conversationState, userState, dialog);
+    }
+
+    /**
+     * Returns a custom Adapter that provides error handling.
+     *
+     * @param configuration The Configuration object to use.
+     * @return An error handling BotFrameworkHttpAdapter.
+     */
+    @Override
+    public BotFrameworkHttpAdapter getBotFrameworkHttpAdaptor(Configuration configuration) {
+        return new AdapterWithErrorHandler(configuration);
+    }
+}
```

Comment:
Just a comment, maybe the `AdapterWithErrorHandler` will be migrated, so this should point to the internal one